lib/storages/data-storage.js

```diff
@@ -4,106 +4,46 @@ const path = require('path');
 const os = require('os');
 const { join } = require('path');
 const { TESTOMAT_TMP_STORAGE_DIR } = require('../constants');
-const { fileSystem, jestHelpers, parseTest } = require('../utils/utils');
+const { fileSystem, parseTest } = require('../utils/utils');
 
-const getTestIdFromTestTitle = parseTest;
+// const getTestIdFromTestTitle = parseTest;
 
 class DataStorage {
   /**
    * Creates data storage instance for specific data type.
    * Stores data to global variable or to file depending on what is applicable for current test runner
-   * (running environment).
    * Recommend to use composition while using this class (instead of inheritance).
    * ! Also the class which will use data storage should be singleton (to avoid data loss).
-   * @param {*} dataType storage type (like 'log', 'artifact'); could be any string, used only to define file path
+   * @param {'log' | 'artifact'} dataType – storage type (like 'log', 'artifact'); could be any string, used only to define file path
+   * @param {Object} options – adapter (e.g. codeceptjs adapter), testId
    */
-  constructor(dataType) {
+  constructor(dataType, { adapter, testId }) {
+    this.adapter = adapter;
+    this.testId = testId;
     this.dataType = dataType || 'data';
     this.isFileStorage = true;
-    this.#refreshStorageType();
+    if (this.adapter === 'playwright' && this.dataType === 'log') this.isFileStorage = false;
```

Comment:
data storage should not know anything about adapter 
it's adapter should configure datastorage